mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java

```diff
@@ -106,74 +105,71 @@ public void update(CRUDOperationContext ctx,
         int numMatched = 0;
         int numFailed = 0;
         BsonMerge merge = new BsonMerge(md);
+        BulkUpdateCtx bulkCtx=new BulkUpdateCtx(collection);
         try {
             ctx.getFactory().getInterceptors().callInterceptors(InterceptPoint.PRE_CRUD_UPDATE_RESULTSET, ctx);
             cursor = collection.find(query, null);
```

Comment:
How many times will we try? Until it is successful?

I think this can be better handled with the solution of the transaction
demarcation problem.

On Tue, Aug 9, 2016 at 2:01 PM, Alec Henninger <notifications@github.com>
wrote:

> In mongo/src/main/java/com/redhat/lightblue/mongo/crud/
> IterateAndUpdate.java
> <https://github.com/lightblue-platform/lightblue-mongo/pull/299#discussion_r74133314>
> :
>
> > @@ -106,74 +105,71 @@ public void update(CRUDOperationContext ctx,
> >          int numMatched = 0;
> >          int numFailed = 0;
> >          BsonMerge merge = new BsonMerge(md);
> > +        BulkUpdateCtx bulkCtx=new BulkUpdateCtx(collection);
> >          try {
> >              ctx.getFactory().getInterceptors().callInterceptors(InterceptPoint.PRE_CRUD_UPDATE_RESULTSET, ctx);
> >              cursor = collection.find(query, null);
>
> What would it take to be able to catch concurrent mods and retry from this
> point, constraining the retry-find to only those document ids which had
> concurrent mods? To me this would be a killer feature (essentially it would
> make very expressive updates appear atomic), but I realize maybe hard to
> implement. Is this something worth considering, if only long term? Or would
> it be a bad idea?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lightblue-platform/lightblue-mongo/pull/299/files/4445c9305f6b8f7a6c756db712b5874118e095f2#r74133314>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADgDDS4tB2qGfLMQMZtZ7Zci6nZg3LQvks5qeNyPgaJpZM4JfOdd>
> .
>